src/kernel-install/90-loaderentry.install

```diff
@@ -22,68 +22,53 @@ COMMAND="$1"
 KERNEL_VERSION="$2"
 ENTRY_DIR_ABS="$3"
 KERNEL_IMAGE="$4"
-INITRD_OPTIONS_START="5"
+INITRD_OPTIONS_SHIFT=4
 
-if ! [[ $KERNEL_INSTALL_MACHINE_ID ]]; then
-    exit 0
-fi
-
-if [ "$KERNEL_INSTALL_LAYOUT" != "bls" ]; then
-    exit 0
-fi
+[ "$KERNEL_INSTALL_LAYOUT" = "bls" ] || exit 0
 
 MACHINE_ID="$KERNEL_INSTALL_MACHINE_ID"
 BOOT_ROOT="$KERNEL_INSTALL_BOOT_ROOT"
 
 BOOT_MNT="$(stat -c %m "$BOOT_ROOT")"
-if [[ "$BOOT_MNT" == '/' ]]; then
+if [ "$BOOT_MNT" = '/' ]; then
     ENTRY_DIR="$ENTRY_DIR_ABS"
 else
     ENTRY_DIR="${ENTRY_DIR_ABS#$BOOT_MNT}"
 fi
 
-if [[ $COMMAND == remove ]]; then
-    rm -f "$BOOT_ROOT/loader/entries/$MACHINE_ID-$KERNEL_VERSION.conf"
-    rm -f "$BOOT_ROOT/loader/entries/$MACHINE_ID-$KERNEL_VERSION+"*".conf"
-    exit 0
-fi
-
-if ! [[ $COMMAND == add ]]; then
-    exit 1
-fi
-
-if ! [[ $KERNEL_IMAGE ]]; then
-    exit 1
-fi
+case "$COMMAND" in
+    remove)
+        exec rm -f \
+            "$BOOT_ROOT/loader/entries/$MACHINE_ID-$KERNEL_VERSION.conf" \
+            "$BOOT_ROOT/loader/entries/$MACHINE_ID-$KERNEL_VERSION+"*".conf"
```

Comment:
It doesn't. It would. It does. They are.

Reply:
Can you guarantee that? :) 

A quick look at coreutils tells me that `-f` only makes `rm` non-interactive, disables prompts and ignores missing files, besides, the actual `rm` function doesn't even know it is being forced. There's plenty of other sources of error though and a quick test case shows us:
```
$ touch foo
$ chattr +i foo 
$ rm -f foo 
rm: cannot remove 'foo': Operation not permitted
$ echo $?
1
```
So your assertion doesn't hold. While this example is artificial, there might be other reasons for `rm` to fail, e.g. a full filesystem.

Reply:
Yes, and in that case 90- should fail because it failed to actually remove.

Reply:
Sure, this might be the correct behaviour, I haven't thought about this yet and happily leave this to others, I just wanted to highlight that in the first place this *is* a behavioural change.

Reply:
I think this change is fine, but please mention it in the commit message.

Reply:
done